modules/flink/src/main/java/org/apache/ignite/sink/flink/IgniteSink.java

```diff
@@ -0,0 +1,231 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ignite.sink.flink;
+
+import org.apache.flink.streaming.api.functions.sink.RichSinkFunction;
+import org.apache.ignite.Ignite;
+import org.apache.ignite.IgniteDataStreamer;
+import org.apache.ignite.IgniteException;
+import org.apache.ignite.IgniteLogger;
+import org.apache.ignite.Ignition;
+import org.apache.ignite.internal.util.typedef.internal.A;
+
+import java.util.Map;
+
+/**
+ * Apache Flink Ignite sink implemented as a RichSinkFunction.
+ */
+public class IgniteSink<IN> extends RichSinkFunction<IN> {
+    /** Default flush frequency. */
+    private static final long DFLT_FLUSH_FREQ = 10000L;
+
+    /** Logger. */
+    private final IgniteLogger log;
+
+    /** Automatic flush frequency. */
+    private long autoFlushFrequency = DFLT_FLUSH_FREQ;
+
+    /** Enables overwriting existing values in cache. */
+    private boolean allowOverwrite = false;
+
+    /** Flag for stopped state. */
+    private static volatile boolean stopped = true;
+
+    /**Ignite grid configuration file. */
+    private static String igniteCfgFile;
+
+    /** Cache name. */
+    private static String cacheName;
+
+    /**
+     * Gets the {@link IgniteDataStreamer} streamer.
+     *
+     * @return {@link IgniteDataStreamer} streamer.
+     */
+    private static IgniteDataStreamer getStreamer() {
+        return SinkContext.getStreamer();
+    }
+
+    /**
+     * Gets the cache name.
+     *
+     * @return Cache name.
+     */
+    public String getCacheName() {
+        return cacheName;
+    }
+    /**
+     * Sets the cache name.
+     *
+     * @param cacheName Cache name.
+     */
+    public void setCacheName(String cacheName) {
+        this.cacheName = cacheName;
+    }
+
+    /**
+     * Gets Ignite configuration file.
+     *
+     * @return Configuration file.
+     */
+    public String getIgniteConfigFile() {
+        return igniteCfgFile;
+    }
+
+    /**
+     * Specifies Ignite configuration file.
+     *
+     * @param igniteConfigFile Ignite config file.
+     */
+    public void setIgniteConfigFile(String igniteConfigFile) {
+        this.igniteCfgFile = igniteConfigFile;
+    }
```

Comment:
looks like setting the configuration file at the constructor is sufficient. Any reason to re/set it later?